tests/node/test_wakunode_filter.nim

```diff
@@ -0,0 +1,109 @@
+{.used.}  
+
+import
+  std/[options, tables, sequtils],
+  stew/shims/net as stewNet,
+  testutils/unittests,
+  chronos,
+  chronicles,
+  os,
+  libp2p/peerstore,
+  libp2p/crypto/crypto
+
+import
+  ../../../waku/waku_core,
+  ../../../waku/node/peer_manager,
+  ../../../waku/node/waku_node,
+  ../../../waku/waku_filter_v2,
+  ../../../waku/waku_filter_v2/client,
+  ../../../waku/waku_filter_v2/subscriptions,
+  ../testlib/common,
+  ../testlib/wakucore,
+  ../testlib/wakunode,
+  ../testlib/testasync,
+  ../testlib/futures
+
+let FUTURE_TIMEOUT = 1.seconds
+
+suite "Full Node - Waku Filter - End to End":
+  var client {.threadvar.}: WakuNode
+  var clientPeerId {.threadvar.}: PeerId
+  var server {.threadvar.}: WakuNode
+  var serverRemotePeerInfo {.threadvar.}: RemotePeerInfo
+  var pubsubTopic {.threadvar.}: PubsubTopicy
+  var contentTopic {.threadvar.}: ContentTopic
+  var contentTopicSeq {.threadvar.}: seq[ContentTopic]
+  var pushHandlerFuture {.threadvar.}: Future[(string, WakuMessage)]
+  var messagePushHandler {.threadvar.}: FilterPushHandler
+  
+  asyncSetup:
+    pushHandlerFuture = newFuture[(string, WakuMessage)]()
+    messagePushHandler = proc(
+      pubsubTopic: PubsubTopic, message: WakuMessage
+    ): Future[void] {.async, closure, gcsafe.} =
+      pushHandlerFuture.complete((pubsubTopic, message))
+
+    pubsubTopic = DefaultPubsubTopic
+    contentTopic = DefaultContentTopic
+    contentTopicSeq = @[DefaultContentTopic]
+
+    let
+      serverKey = generateSecp256k1Key()
+      clientKey = generateSecp256k1Key()
+
+    server = newTestWakuNode(serverKey, ValidIpAddress.init("0.0.0.0"), Port(0))
+    client = newTestWakuNode(clientKey, ValidIpAddress.init("0.0.0.0"), Port(0))
+
+    waitFor allFutures(server.start(), client.start())
+
+    waitFor server.mountFilter()
+    waitFor client.mountFilterClient()
+
+    client.wakuFilterClient.registerPushHandler(messagePushHandler)
+    serverRemotePeerInfo = server.peerInfo.toRemotePeerInfo()
+    clientPeerId = client.peerInfo.toRemotePeerInfo().peerId
+
+  asyncTeardown:
+    waitFor allFutures(client.stop(), server.stop())
+
+  asyncTest "Full Client Node to Full Service Node Subscription":
+    # When a full client node subscribes to a full service node
+    let subscribeResponse = await client.filterSubscribe(
+      some(pubsubTopic), contentTopicSeq, serverRemotePeerInfo
+    )
+
+    # Then the subscription is successful
+    check:
+      subscribeResponse.isOk()
+      server.wakuFilter.subscriptions.len == 1
```

Comment:
```suggestion
    
    assert subscribeResponse.isOk(), $subscribeResponse.error
    check server.wakuFilter.subscriptions.len == 1
```
If possible, I'd apply that elsewhere.

Reply:
Indeed! Must've slipped my mind. I'm planning to do a "standardising" PR after I finish the current one I'm working on to mend all places I skipped.